File: demo/java/src/main/java/info/chenliang/tank/Client2ServerStub.java
package info.chenliang.tank;
import java.io.*;
import info.chenliang.talky.*;
public abstract class Client2ServerStub {
protected abstract DataOutputStream begin();
protected abstract void end();
public void fire(byte playerId) throws Exception{
DataOutputStream dos = begin();
dos.write(0);
dos.writeByte(playerId);
end();
}
public void useItem(byte seq) throws Exception{
DataOutputStream dos = begin();
dos.write(1);
dos.writeByte(seq);
end();
}
}

File: demo/java/src/main/java/info/chenliang/tank/Client2ServerStub.cs
namespace info.chenliang.tank{
using System;
using System.IO;
using MiscUtil;
using MiscUtil.IO;
using MiscUtil.Conversion;
using Talky;
public abstract class Client2ServerStub {
protected abstract MemoryEndianBinaryWriter Begin();
protected abstract void End();
public void fire(sbyte playerId){
MemoryEndianBinaryWriter writer = Begin();
writer.WriteByte(0);
writer.WriteInt8(playerId);
End();
}
public void useItem(sbyte seq){
MemoryEndianBinaryWriter writer = Begin();
writer.WriteByte(1);
writer.WriteInt8(seq);
End();
}
}
}
